core/primitives/src/utils.rs

```diff
@@ -340,7 +337,7 @@ fn create_hash_upgradable(
     block_height: BlockHeight,
     salt: u64,
 ) -> CryptoHash {
-    if protocol_version < CREATE_HASH_PROTOCOL_VERSION {
+    if !ProtocolFeature::CreateHash.enabled(protocol_version) {
```

Comment:
Could you change the `create_hash_upgradable` function to take a `ProtocolFeature` as input instead of `ProtocolVersion` as the first parameter? Based on the new parameter, instead of `if !ProtocolFeature::CreateHash.enabled(protocol_version)`, could we convert it to a `if let` style check?

Reply:
(aside) @nagisa giving it a second shot!